queue/driver_mongo.go

```diff
@@ -119,43 +112,53 @@ func (d *mongoDriver) ID() string {
 	return d.instanceID
 }
 
+// Open opens a new owned connection for the driver. If a connection has already
+// been previously established for the driver, this is a no-op.
 func (d *mongoDriver) Open(ctx context.Context) error {
-	if d.canceler != nil {
+	d.mu.RLock()
+	if d.cancel != nil {
+		d.mu.RUnlock()
 		return nil
 	}
+	d.mu.RUnlock()
 
 	client, err := mongo.Connect(ctx, options.Client().ApplyURI(d.opts.URI))
 	if err != nil {
-		return errors.Wrapf(err, "problem opening connection to mongodb at '%s", d.opts.URI)
+		return errors.Wrapf(err, "opening connection to mongodb at '%s", d.opts.URI)
 	}
+	d.ownsClient = true
```

Comment:
Should the lock be obtained before updating this?

Reply:
Yeah, you're right. I moved this statement down to the lock in `start` to ensure that both `client` and `ownsClient` are set atomically.